File: 2839.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int sum = 0;
        while (n > 0) {
            if (n % 5 == 0) {
                sum++;
                n -= 5;
            } else if (n % 3 == 0) {
                sum++;
                n -= 3;
            } else if (n > 5) {
                n -= 5;
                sum++;
            } else {
                sum = -1;
                break;
            }
        }
        System.out.println(sum);
    }
}

File: 2839.py
n = int(input())
sum = 0
while n > 0:
    if n % 5 == 0:
        sum += 1
        n -= 5
    elif n % 3 == 0:
        sum += 1
        n -= 3
    elif n > 5 :
        n -= 5
        sum += 1
    else:
        sum = -1
        break
print(sum)